arbnode/mel/extraction/message_extraction_function_test.go

```diff
@@ -0,0 +1,378 @@
+			melStateParentHash: prevParentBlockHash,
+			lookupBatches:      emptyLookupBatches,          // 0 batches
+			lookupDelayedMsgs:  successfulLookupDelayedMsgs, // 1 batch posting report
+			expectedError:      "batch posting reports 1 do not match the number of batches 0",
+		},
+		{
+			name:               "batch serialization fails",
+			melStateParentHash: prevParentBlockHash,
+			lookupBatches:      successfulLookupBatches,
+			lookupDelayedMsgs:  successfulLookupDelayedMsgs,
+			serializer:         failingSerializer,
+			expectedError:      "serialization error",
+		},
+		{
+			name:               "parsing batch posting report fails",
+			melStateParentHash: prevParentBlockHash,
+			lookupBatches:      successfulLookupBatches,
+			lookupDelayedMsgs:  successfulLookupDelayedMsgs,
+			serializer:         emptySerializer,
+			parseReport:        failingParseReport,
+			expectedError:      "batch posting report parsing error",
+		},
+		{
+			name:               "mismatched batch posting report batch hash and actual batch hash",
+			melStateParentHash: prevParentBlockHash,
+			lookupBatches:      successfulLookupBatches,
+			lookupDelayedMsgs:  successfulLookupDelayedMsgs,
+			serializer:         emptySerializer,  // Returns nil, hash will be different from parseReport
+			parseReport:        emptyParseReport, // Returns empty hash
+			expectedError:      "batch data hash incorrect",
+		},
+		{
+			name:               "parse sequencer message fails",
+			melStateParentHash: prevParentBlockHash,
+			lookupBatches:      successfulLookupBatches,
+			lookupDelayedMsgs:  successfulLookupDelayedMsgs,
+			serializer:         successfulSerializer,
+			parseReport:        successfulParseReport,
+			parseSequencerMsg:  failingParseSequencerMsg,
+			expectedError:      "failed to parse sequencer message",
+		},
+		{
+			name:                 "extracting batch messages fails",
+			melStateParentHash:   prevParentBlockHash,
+			lookupBatches:        successfulLookupBatches,
+			lookupDelayedMsgs:    successfulLookupDelayedMsgs,
+			serializer:           successfulSerializer,
+			parseReport:          successfulParseReport,
+			parseSequencerMsg:    successfulParseSequencerMsg,
+			extractBatchMessages: failingExtractBatchMessages,
+			expectedError:        "failed to extract batch messages",
+		},
+		{
+			name:                 "OK",
+			melStateParentHash:   prevParentBlockHash,
+			lookupBatches:        successfulLookupBatches,
+			lookupDelayedMsgs:    successfulLookupDelayedMsgs,
+			serializer:           successfulSerializer,
+			parseReport:          successfulParseReport,
+			parseSequencerMsg:    successfulParseSequencerMsg,
+			extractBatchMessages: successfulExtractBatchMessages,
+			expectedMsgCount:     2,
+			expectedDelayedSeen:  1,
+			expectedMessages:     2,
+			expectedDelayedMsgs:  1,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			block := createBlock()
```

Comment:
You create a block here, but then only use it to hold a block.Header()
Couldn't you just have the header?